xbmc/pvr/PVRDatabase.cpp

```diff
@@ -625,31 +624,25 @@ bool CPVRDatabase::Persist(CPVRChannel &channel)
         channel.UniqueID(), (channel.IsRadio() ? 1 :0), (channel.IsHidden() ? 1 : 0), (channel.IsUserSetIcon() ? 1 : 0), (channel.IsUserSetName() ? 1 : 0), (channel.IsLocked() ? 1 : 0),
         channel.IconPath().c_str(), channel.ChannelName().c_str(), 0, (channel.EPGEnabled() ? 1 : 0), channel.EPGScraper().c_str(), channel.LastWatched(), channel.ClientID(),
         channel.EpgID());
+
+    QueueInsertQuery(strQuery);
   }
   else
   {
     /* update channel */
-    strQuery = PrepareSQL("REPLACE INTO channels ("
-        "iUniqueId, bIsRadio, bIsHidden, bIsUserSetIcon, bIsUserSetName, bIsLocked, "
-        "sIconPath, sChannelName, bIsVirtual, bEPGEnabled, sEPGScraper, iLastWatched, iClientId, "
-        "idChannel, idEpg) "
-        "VALUES (%i, %i, %i, %i, %i, %i, '%s', '%s', %i, %i, '%s', %u, %i, %i, %i)",
+    strQuery = PrepareSQL("UPDATE channels SET "
+        "iUniqueId = %i, bIsRadio = %i, bIsHidden = %i, bIsUserSetIcon = %i, bIsUserSetName = %i, bIsLocked = %i, "
+        "sIconPath= %s, sChannelName = %s, bIsVirtual = %i, bEPGEnabled = %i, sEPGScraper = %s, iLastWatched = %u, "
+        "iClientId = %i, idChannel = %i, idEpg = %i "
+        "WHERE iUniqueId = %i AND iClientId = %i",
         channel.UniqueID(), (channel.IsRadio() ? 1 :0), (channel.IsHidden() ? 1 : 0), (channel.IsUserSetIcon() ? 1 : 0), (channel.IsUserSetName() ? 1 : 0), (channel.IsLocked() ? 1 : 0),
         channel.IconPath().c_str(), channel.ChannelName().c_str(), 0, (channel.EPGEnabled() ? 1 : 0), channel.EPGScraper().c_str(), channel.LastWatched(), channel.ClientID(),
-        channel.ChannelID(),
-        channel.EpgID());
-  }
+        channel.ChannelID(), channel.EpgID(), channel.UniqueID(), channel.ClientID());
```

Comment:
no need to update iUniqueId and iClientId as they are part of the where condition.